Update success.tpl CSS (pre/post upgrade messages)

mlutfy opened the pull request:
Overview
----------------------------------------

Initially I just wanted to add a link to the "CiviCRM System Status", because post-upgrade, I think that's the first thing people should check.

However, after adding the link, it was ugly, just felt like I was adding more clutter, so I went down a rabbit hole attempt to cleanup a bit.

- Pre-upgrade: Hides the "cancel" button, because it's clutter and people know how to use a back button
- Bigger font-size, sans-serif
- Less bold, because it's annoying; replaced by even bigger text, when relevant
- Align icons (mostly on the post-upgrade messages)

Before
----------------------------------------

Before running the upgrade:

![civicrm-preup-before](https://user-images.githubusercontent.com/254741/228918088-bc337f49-0713-43f2-b6a7-686f11a1edbb.png)

After running the upgrade:

![civicrm-finished-before](https://user-images.githubusercontent.com/254741/228918151-b3df24c9-19ff-4720-8091-cc4e94d6af1a.png)

After
----------------------------------------

Before running the upgrade:

![civicrm-preup-after-warnings](https://user-images.githubusercontent.com/254741/228918262-81e57517-1930-4827-b442-fa70d27174cd.png)

Before running the upgrade (if there are no upgrade warnings):

![civicrm-preup-after-nowarnings](https://user-images.githubusercontent.com/254741/228918260-9baebd15-7a11-4490-849a-7ebeb5905363.png)

After running the upgrade:

![civicrm-finished-after](https://user-images.githubusercontent.com/254741/228918263-5684b9c8-7b58-4ab8-9844-ee050b2c25bb.png)

Comments
----------------------------------

Some of the funnyness around buttons is because of Shoreditch and can be ignored.

I think the post-upgrade message should be simplified with a more clear call-to-action (donate), but it would break translations and I was not in the mood.

mlutfy committed:
Update success.tpl CSS (pre/post upgrade messages)

Removes inline CSS and replaces icons with font-awesome, and attempts
to make the pre/post upgrade look a bit nicer.

Updates the post-upgrade thank you message and also adds a link to
encourage admin to view the System Check.

eileenmcnaughton commented:
@vingle @artfulrobot any input?

demeritcowboy commented on templates/CRM/common/success.tpl:
```diff
@@ -12,42 +12,73 @@
+  <div class="crm-container" style="background-color: #FFFFE3; border: 1px #F8FF00 solid;">
+    <div class="crm-success-wrapper">
+      <form method="post">
+        {if $preUpgradeMessageX}
```
X?

eileenmcnaughton replied:
xx

mlutfy replied:
fixed :see_no_evil: 
that was me testing what happens when there are no upgrade messages ;)

demeritcowboy commented:
Other than the mysterious "X" which displays an undefined variable error instead of the preupgrade message, this looks nice. +1

demeritcowboy commented:
One more note: If this is merged, the hardcoded `color: white` should be removed from [General.php](https://github.com/civicrm/civicrm-core/blob/e996efbcb401ec8533c6c2f406bb50347ba65485/CRM/Upgrade/Incremental/General.php#L181) since now if you previously visited that `<a>` link, which you may well have, it appears white on light yellow, and is almost invisible.

mattwire commented:
@mlutfy Some good feedback from @nicol and @demeritcowboy. If you can address those things I'll merge it quickly before we start swapping wheels.

mlutfy commented:
Anyone know why this page displays with the frontend theme in Drupal?

The frontend themes usually don't support many of the backend themes (ex: alerts, maybe font-awesome).

vingle commented:
@mlutfy – ah I see. It does load on WordPress and Joomla. Font Awesome loads on my Drupal 9 upgrade screen tho, so that should be safe.  

As you're adding that CSS just to support Drupal, what about instead adding a class that Drupal should recognise like `messages messages--warning` and `messages messages--status` to get the Drupal alert style? ie

```
<div class="crm-container">
 <div class="message messages--warning status">
  This messages will have either Drupal's yellow warning style or Civi's
  </div>
</div>
<div class="crm-container">
 <div class="message messages--status help">
  This messages will have either Drupal's green warning style or Civi's
  </div>
</div>
```

mlutfy commented:
@vingle Thanks for the suggestions. I tried to implement some of the suggestions, to the extent that this is a frontend form and I think we should render as a backend form before doing more changes, but I went way too far down the rabbit hole. I'd rather not add Drupal-specific classes, I just wanted to avoid random visual results depending on the CMS. I removed as much CSS as I could, and hopefully we can remove most of it when we flip to a backend form.

@demeritcowboy I removed the white link from General.php, but it's a b it difficult to test. Since I was breaking the translations for this string, I fixed the ts usage.

I also decided to break/fix that "thank you" string, because it was really annoying me.

Here is what it looks like on Drupal9 with the new changes (I replaced `em` with `rem`, but I'm not too familiar with those differences, so I improvised):

![image](https://user-images.githubusercontent.com/254741/229119544-bb4b7e09-6f4d-49ed-b548-d0c6d0b0d365.png)

I don't know the logo will be missed, but I figure that simpler is better, and I still think that message should be simplified, and keeping that out of scope for now:

![image](https://user-images.githubusercontent.com/254741/229123868-b7c7f4b7-d218-417a-8340-3ecf15273ca4.png)

This is after fixing the "Thank you" message:

![image](https://user-images.githubusercontent.com/254741/229130974-902a28ed-d23d-4332-9ef9-59a74c232bdc.png)

While grave-digging, found some possibly un-used things, but wasn't sure:

- `templates/CRM/common/upgradeCleanup.tpl`
- in `css/admin.css` there is `.upgrade-success` that is only used by upgradeCleanup - that CSS file is mostly rubbish and I'd rather avoid it, even if the upgrade screen uses it

demeritcowboy commented:
Something's wrong with the donation link. I think it's because it's now in single quotes - checking...

re: frontend theme: https://github.com/civicrm/civicrm-drupal/blob/0ec6bf7886d03205b0f4f74533357525efecb552/modules/civicrmtheme/civicrmtheme.module#L91

vingle commented:
> re: frontend theme: https://github.com/civicrm/civicrm-drupal/blob/0ec6bf7886d03205b0f4f74533357525efecb552/modules/civicrmtheme/civicrmtheme.module#L91 

Is that a way to adjust this? Maybe it would be better to deal with the quirk of Drupal viewing a backend form as a front-end form, rather than making every CMS load styles that clash with any non-Grenwich Civi theme that's running? The hard-baked styles will also clash with whatever comes after Greenwich…

demeritcowboy commented:
My guess is that it was there because during an upgrade you want as little as possible code running to avoid things like chicken-and-egg situations, but if other cmses have already been running for a long time with the civi theme during upgrade then it doesn't seem a big risk. And it seems drupal 9 doesn't have that `if` clause.

mlutfy commented:
jenkins, test this please

mlutfy commented:
@demeritcowboy Thanks! - fixed the links, I think?

There are weird test fails:

```
[[Download build-1 (type 'drupal-clean' in '/home/jenkins/bknix-dfl/build/build-1')]]
[[Update caches]]
SKIP: git_cache_setup 'https://github.com/civicrm/civivolunteer.git' /home/jenkins/bknix-dfl/build/.civibuild/cache/civicrm/civivolunteer.git' (recently updated; ttl=60)
SKIP: git_cache_setup 'https://lab.civicrm.org/extensions/angularprofiles.git' /home/jenkins/bknix-dfl/build/.civibuild/cache/ginkgostreet/org.civicrm.angularprofiles.git' (recently updated; ttl=60)
error: could not lock config file config: File exists
fatal: could not set 'remote.origin.url' to 'https://lab.civicrm.org/extensions/civirules.git'
```

(gitlab seems to be working fine, and I checked fail2ban/firewall)

And thanks for the pointer in civicrmtheme. I found this commit:  
https://github.com/civicrm/civicrm-svn/commit/e5cd2cba519f53a14db5e630838d31923ba3849f

which leads to: https://issues.civicrm.org/jira/browse/CRM-10612 - but not very detailed. It was on Drupal7, 4.1 to 4.2 upgrade.

I was testing on Drupal9 though. And as you say, on WordPress this does not apply, and CiviCRM are less brutal than before, so I would be curious to try removing that code, but separate PR/repo.

mlutfy commented:
jenkins, test this please

mlutfy commented:
For what it's worth, I tested WP too:

![image](https://user-images.githubusercontent.com/254741/229191014-7fd46046-6838-481f-8049-9bd0727c294f.png)

demeritcowboy commented:
Have put merge-ready.

I don't know why I didn't see it but drupal 9 does also have the check for upgrade so [returns early](https://github.com/civicrm/civicrm-drupal-8/blob/143bed9fe10c02271a0bcaa4f4f4c2e527f04a96/modules/civicrmtheme/src/Theme/CivicrmThemeNegotiator.php#L70). But can treat as a followup if desired.

mlutfy commented:
Thanks! I opened PRs on the Drupal repos to change that behaviour:

- https://github.com/civicrm/civicrm-drupal-8/pull/81
- https://github.com/civicrm/civicrm-drupal/pull/665

demeritcowboy commented:
@mlutfy What order did you want to do this in? I'm ok to merge the other PRs, in which case would you then update this or should we also merge this and leave further updates for now?

eileenmcnaughton commented:
Would it be possible to add a comment to success.tpl explaining why putting css in the template is acceptable in this situation?

mlutfy commented:
@eileenmcnaughton Yep, comment added.

@demeritcowboy PRs Can be merged independently. I'll circle back at some point to cleanup the CSS. (the Drupal changes seem harmless to me, but who knows, maybe it'll need to be reverted).

demeritcowboy commented:
I'll merge the other ones. I'm not sure what you just changed here but something is off now (drupal 7). It really wants to warn me about something:

<img width="352" alt="Untitled3" src="https://user-images.githubusercontent.com/2967821/230108421-7554fa40-fc2e-40c6-9536-d02fd9382b4d.png">

mlutfy commented:
Is that the new Montreal theme? That looks like a typical road in downtown Montreal during construction season (i.e. summer).

 I'll circle back and get rid of the CSS and do more testing :)

mlutfy commented:
I did some minor updates and more testing on all CMSes:

- Replaces the old CiviCRM status/checkbox icons with font-awesome (as suggested by @nicol)
- Avoid using generic `success` or `warning` classes because they conflict with the Drupal theme (avoid Montreal-construction-style visual)

I could not find suitable generic CSS classes from CiviCRM to avoid inline CSS. There are classes for alerts (that appear in the corner of the screen), but they do not apply to full-screen messages. I updated the comment to that effect.

Drupal7 with Seven and Shoreditch, pre/post upgrade:

![image](https://user-images.githubusercontent.com/254741/232252107-09c49f05-23b0-45eb-a7fa-f249fef90095.png)

![image](https://user-images.githubusercontent.com/254741/232308630-844509d5-c1d2-4436-bfd7-6f3d7f4cdaa2.png)

WordPress pre/post-upgrade:

![image](https://user-images.githubusercontent.com/254741/232309032-36bc040e-0b4f-433b-bdec-3312f84da030.png)

![image](https://user-images.githubusercontent.com/254741/232308971-630be2fb-f614-480b-b565-2af0c30215c6.png)

Without Shoreditch:

![image](https://user-images.githubusercontent.com/254741/232309163-37c59281-bf7e-4467-969b-8b8bab9f1168.png)

![image](https://user-images.githubusercontent.com/254741/232309192-a08dd068-45bd-48c2-a2f4-c9507f53d697.png)

Drupal9:

![image](https://user-images.githubusercontent.com/254741/232309781-0a6c699f-a97d-423e-8d5d-e3e25325b72b.png)

![image](https://user-images.githubusercontent.com/254741/232309819-cc2c6e01-88f2-47ba-b19b-aecf8c789a27.png)

vingle commented:
@mjw fair enough - good reasons. Tho it's a pattern used pretty widely across Civi, so at the point at which we are able to replace all `<div class="status">` with `<div class="crm-alert-warning">` matching the bad pattern here will make it easier to find later ;).

The current focus for myself and @artfulrobot for a replacement to the Style Guide is [Theme Test](https://lab.civicrm.org/extensions/themetest). I've added [an issue](https://github.com/civicrm/civicrm-core/pull/25961#issuecomment-1511494184) to add some snippets for current Alert handling.

demeritcowboy commented:
I gave it a last little run and it sounds like it's more or less agreed to leave more tweaking until later.

I notice the donation link still has `src=ug` which is left over from before when it was a different link, but it doesn't hurt anything. It could still be useful to have even if not being tracked.
